exporter/src/main/scala/uk/gov/nationalarchives/consignmentexport/KeycloakClient.scala

```diff
@@ -0,0 +1,42 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import org.keycloak.OAuth2Constants
+import org.keycloak.admin.client.resource.{RealmResource, UsersResource}
+import org.keycloak.admin.client.{Keycloak, KeycloakBuilder}
+import org.keycloak.representations.idm.UserRepresentation
+import sttp.client.{HttpURLConnectionBackend, Identity, NothingT, SttpBackend}
+import uk.gov.nationalarchives.consignmentexport.Config.Configuration
+
+import scala.language.postfixOps
+
+class KeycloakClient(keycloakAdminClient: Keycloak, config: Configuration) {
+  private def realmResource(client: Keycloak): RealmResource = client.realm(config.auth.realm)
+  private def usersResource(realm: RealmResource): UsersResource = realm.users()
+
+  def getUserDetails(userId: String): UserRepresentation = {
+    val realm = realmResource(keycloakAdminClient)
+    val users = usersResource(realm)
+
+    try {
+      val userResource = users.get(userId)
+      userResource.toRepresentation
```

Comment:
The Keycloak library appears to be doing a synchronous HTTP call here. It looks like that's just how it works - there's no async option. @MancunianSam - how does that fit into the Cats IO approach? It "works" without using IO, but are there advantages to wrapping it an IO so that it gets evaluated along with the rest of the asynchronous IO?